### PR TITLE
Token swap

### LINKED ISSUE #34: Vault client security
Currently we use a vault token from the environment to authenticate against Vault for the Astral service. Should this have additional security? For example:
1) instead configure authentication tokens which we exchange for a new operational token?
2) mTLS between astral and vault? where do these certs come from?

### PR DESCRIPTION
**Pull request by suprjinx**
Switch the Vault token after boostrapping the engines
Fixes #34

### CODE REVIEW

File: app/lib/clients/vault/policy.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module Clients
+  class Vault
+    class << self
+      def rotate_token
```
Comment (GeorgeJahad) [inline app/lib/clients/vault/policy.rb:4]: Is it worth adding a test for rotate_token?

**Review by GeorgeJahad — DISMISSED** [review]
I left a comment, but lgtm.

File: app/lib/clients/vault/policy.rb (hunk 1 above)
Comment (suprjinx) [reply app/lib/clients/vault/policy.rb:4]: well, yes!

Comment (suprjinx) [reply app/lib/clients/vault/policy.rb:4]: added a test case in `clients/vault_test.rb`

File: test/lib/clients/vault_test.rb [hunk 2]
```diff
@@ -18,15 +20,15 @@ class VaultTest < ActiveSupport::TestCase
...
+    @client.stub :kv_mount, kv_mount do
       assert @client.configure_kv
       engines = vault_client.sys.mounts
-      assert_equal "kv", engines[intermediate_ca_mount.to_sym].type
+      assert_equal "kv", engines[kv_mount.to_sym].type
     end
   end
 
-  test "#configure_pki" do
+  test ".configure_pki" do
```
Comment (GeorgeJahad) [inline test/lib/clients/vault_test.rb:31]: what is the diff between '#' and '.'?

is there convention here that I should understand?

**Review by GeorgeJahad — DISMISSED** [review]
lgtm

File: test/lib/clients/vault_test.rb (hunk 2 above)
Comment (suprjinx) [reply test/lib/clients/vault_test.rb:31]: It's a convention from `rspec` which I adopted for the descriptions here -- https://grosser.it/2010/06/18/how-to-name-rspec-describe-blocks-for-methods/

**Review by GeorgeJahad — APPROVED** [review]

**Pull request merged**